src/commands/generateFunctions.ts

```diff
@@ -0,0 +1,104 @@
+import { ApiTreeItem } from "../explorer/ApiTreeItem";
+import { OpenApiEditor } from "../explorer/editors/openApi/OpenApiEditor";
+import { ext } from "../extensionVariables";
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { localize } from "../localize";
+import { cpUtils } from "../utils/cpUtils";
+
+// tslint:disable: no-non-null-assertion
+export async function generateFunctions(node?: ApiTreeItem): Promise<void> {
+    if (!node) {
+        node = <ApiTreeItem>await ext.tree.showTreeItemPicker(ApiTreeItem.contextValue);
+    }
+
+    ext.outputChannel.show();
+
+    const openApiEditor: OpenApiEditor = new OpenApiEditor();
+    const openAPIDocString = await openApiEditor.getData(node);
+
+    const languages = ['Python', 'CSharp', 'TypeScript', 'Java'];
+    const language = await ext.ui.showQuickPick(
+        languages.map((s) => {return { label: s, description: '', detail: '' }; }), { placeHolder: "Select language", canPickMany: false});
+
+    const uris = await askFolder();
+
+    const openAPIFilePath = path.join(uris[0].fsPath, `${node!.apiContract.name}.json`);
+
+    await window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("openAPI", `Downloading OpenAPI Document for API '${node.apiContract.name}'...`),
+            cancellable: false
+        },
+        async () => {
+            await fse.writeFile(openAPIFilePath, openAPIDocString);
+        }
+    ).then(async () => {
+        window.showInformationMessage(localize("openAPIDownloaded", `Downloaded OpenAPI Document for API '${node!.apiContract.name} successfully.`));
+    });
+
+    const autoRestInstalled = await isAutoRestInstalled();
+    if (!autoRestInstalled) {
+        await window.withProgress(
+            {
+                location: ProgressLocation.Notification,
+                title: localize("installAutorest", `Installing autorest...`),
+                cancellable: false
+            },
+            async () => {
+                await cpUtils.executeCommand(ext.outputChannel, undefined, 'npm install -g autorest');
+            }).then(async () => {
+                window.showInformationMessage(localize("installAutorest", `Autorest installed successfully.`));
+            });
+    }
+
+    await window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("generateFunctions", `Generating functions for API '${node.apiContract.name}'...`),
+            cancellable: false
+        },
+        async () => {
+            const args: string[] = [];
+            args.push(`--input-file:${openAPIFilePath}`);
+            args.push('--use:@autorest/azure-functions@0.0.1-preview-dev.20200727.5');
+            args.push(`--output-folder:${uris[0].fsPath}`);
+            args.push('--no-async');
+            args.push(`--language:${language.label.toLowerCase()}`);
```

Comment:
You'll need to do something like the following, to switch between languages and plugins used.:

```javascript
public async execute(wizardContext: IFunctionWizardContext): Promise<void> {
        const uris = await this.askDocument();
        const uri = uris[0];

        const args: string[] = [];
        args.push(`--input-file:${uri.fsPath} --version:3.0.6314`);

        if (wizardContext.language === ProjectLanguage.TypeScript) {
            args.push('--azure-functions-typescript');
            args.push('--no-namespace-folders:True');
        } else if (wizardContext.language === ProjectLanguage.CSharp) {
            args.push('--namespace:Microsoft.Azure.Stencil');
            args.push('--azure-functions-csharp');
        } else if (wizardContext.language === ProjectLanguage.Java) {
            args.push('--namespace:com.microsoft.azure.stencil');
            args.push('--azure-functions-java');
        } else if (wizardContext.language === ProjectLanguage.Python) {
            args.push('--azure-functions-python');
            args.push('--no-namespace-folders:True');
            args.push('--no-async');
        } else {
            throw new Error(localize("notSupported", "Not a supported language. We currently support C#, Java, Python, and Typescript"));
        }
```

Reference on how VSCode Functions is planning to use it: https://raw.githubusercontent.com/vrdmr/vscode-azurefunctions/8bf23d07eb9576a8b7462d274e112421ff1deb7d/src/commands/createNewProject/openAPISteps/OpenAPICreateStep.ts